### PR TITLE
gundeck: Fix parsing errors for SNS ARN for VOIP Tokens

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/browse/WPB-9021

This also includes returning a specific error when a client attempts to register a VOIP token, as to avoid any accidental inserts.

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/test/unit/Test/Wire/API/MLS.hs [hunk 1]
```diff
@@ -308,7 +308,7 @@ spawn cp minput = do
        in snd <$> concurrently writeInput readOutput
   case (mout, ex) of
     (Just out, ExitSuccess) -> pure out
-    _ -> assertFailure $ "Failed spawning process\n" <> show mout <> "\n" <> show ex
+    _ -> assertFailure "Failed spawning process"
```
Comment (akshaymankar) [inline libs/wire-api/test/unit/Test/Wire/API/MLS.hs:311]:
```suggestion
    _ -> assertFailure $ "Failed spawning process\n" <> show mout <> "\n" <> show ex
```

File: services/gundeck/src/Gundeck/Push.hs [hunk 2]
```diff
@@ -374,38 +375,53 @@ nativeTargets psh rcps' alreadySent =
...
+                ( \a' ->
+                    addrEqualClient a a'
+                      && a ^. addrApp == a' ^. addrApp
+                      && a' ^. addrTransport == t
+                )
+                as
+            )
+        defPreference = case psh ^. pushNativePriority of
+          LowPriority -> ApsStdPreference
+          HighPriority -> ApsVoIPPreference
```
Comment (akshaymankar) [inline services/gundeck/src/Gundeck/Push.hs:402]: Perhaps this shouldn't be reverted.

File: services/brig/docs/swagger-v3.json [hunk 3]
```diff
@@ -14,7 +14,9 @@
       "enum": [
         "GCM",
         "APNS",
-        "APNS_SANDBOX"
+        "APNS_SANDBOX",
+        "APNS_VOIP",
+        "APNS_VOIP_SANDBOX"
```
Comment (akshaymankar) [inline services/brig/docs/swagger-v3.json:19]: Old swagger changed??

Comment (akshaymankar) [reply services/brig/docs/swagger-v3.json:19]: Looks like previously it was wrongly changed. This just fixes it. :shrug:

**Review by elland — APPROVED** [review]

**Pull request merged**